app/src/main/java/com/geeksville/mesh/NodeInfo.kt

```diff
@@ -164,7 +164,7 @@ data class NodeInfo(
     var deviceMetrics: DeviceMetrics? = null,
     var channel: Int = 0,
     var environmentMetrics: EnvironmentMetrics? = null,
-    var hopsAway: Int = 0
+    var hopsAway: Int? = null
```

Comment:
avoid making any changes to `NodeInfo`, this would break the API

Reply:
current behavior remains the same, so there's no loss. and it's a breaking change. with the new nodeDB overhaul the API parcelables should not change at all.

one option is using -1 instead of null in both `NodeEntity` and `NodeInfo`, but whether that adds enough value to justify the added refactor is up to you.

Reply:
[Refactor_hopsAway.PATCH](https://github.com/user-attachments/files/17311997/Refactor_hopsAway.PATCH)

@andrekir - I had a minute to work up a patch to make the `-1` change, also reverting the db change since it's no longer necessary. Give it a look?

Reply:
Just in case we want to not have a(nother) db migration in this pr :sweat_smile:

Reply:
@jamesarich patch looks really good! I'll wait for you to push those.

at a glance I think it's only missing revert the db version bump to 14 in `MeshtasticDatabase` and the Elvis operator (= hopsAway ?: -1,) in `NodeEntity.toNodeInfo()`.

Reply:
@ianmcorvidae only external API clients like the ATAK Plugin would not see null hopsAway.

Reply:
> @jamesarich patch looks really good! I'll wait for you to push those.
> 
> at a glance I think it's only missing revert the db version bump to 14 in `MeshtasticDatabase` and the Elvis operator (= hopsAway ?: -1,) in `NodeEntity.toNodeInfo()`.

@andrekir - applied patch, with those 2 changes - should be good to go 🤞